File: First Semester/Algorithms/1- bubble sorting.py
import random

def generate_random_numbers(num):
    random_numbers = [random.randint(1, 1000) for _ in range(num)]
    return random_numbers

def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]

if __name__ == "__main__":
    num_of_numbers = 500
    random_numbers = generate_random_numbers(num_of_numbers)
    print("Random numbers before sorting:")
    print(random_numbers)

    bubble_sort(random_numbers)

    print("\nRandom numbers after sorting using Bubble Sort:")
    print(random_numbers)

File: First Semester/Algorithms/1- bubble sorting.js
const totalNumber = 1000;

function generateRandomNumbers(num) {
  let arr = [];
  for (let x = 0; x < num; x++) {
    const randomNumber = Math.floor(Math.random() * 100) + 1;
    arr.push(randomNumber);
  }
  return arr;
}

const array = generateRandomNumbers(totalNumber);

console.log("array before bubble sorting:");
console.log(array);

function bubbleSort(array) {
  for (let x = 0; x < array.length; x++) {
    for (let y = 0; y < array.length; y++) {
      if (array[y] > array[y + 1]) {
        let temp = array[y];
        array[y] = array[y + 1];
        array[y + 1] = temp;
      }
    }
  }
  return array;
}

const sortedArray = bubbleSort(array);
console.log("array after bubble sorting:");
console.log(sortedArray);
